core/src/com/unciv/logic/city/City.kt

```diff
@@ -254,6 +254,14 @@ class City : IsPartOfGameInfoSerialization {
         return cityResources
     }
 
+    fun getResourceAmount(resourceName:String): Int {
+        val resource = getRuleset().tileResources[resourceName] ?: return 0
+
+        if (resource.hasUnique(UniqueType.CityResource))
+            return getCityResources().firstOrNull { it.resource == resource }?.amount ?: 0
```

Comment:
If you're getting resources from multiple sources (ie, both improvements and buildings), this only finds one of these sources. You'd want the sum of that resource, not the first listing of it